src/QsCompiler/CompilerOptimizations/Utils.fs

```diff
@@ -0,0 +1,203 @@
+﻿module Microsoft.Quantum.QsCompiler.CompilerOptimization.Utils
+
+open System
+open System.Collections.Generic
+open System.Collections.Immutable
+open System.Numerics
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxExtensions
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+
+
+/// Option default value operator
+let (|?) = defaultArg
+
+
+/// Shorthand for a QsExpressionKind
+type Expr = QsExpressionKind<TypedExpression, Identifier, ResolvedType>
+/// Shorthand for a QsTypeKind
+type TypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>
+
+
+/// Represents the global dictionary that maps names to callables
+type CallableDict = {
```

Comment:
So, what was the intent/purpose of this additional wrapper?

Reply:
CallableDict is much easier to write than ImmutableDictionary<QsQualifiedName, QsCallable>. Also, the type specifically represents the global dictionary the compiler uses to map qualified names to callables in the program, and has a helper function to support that.